pertpy/plot/_mixscape_barplot.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import pandas as pd
+from anndata import AnnData
+from plotnine import (
+    aes,
+    element_text,
+    facet_wrap,
+    geom_bar,
+    ggplot,
+    labs,
+    scale_fill_manual,
+    theme,
+    theme_classic,
+    xlab,
+    ylab,
+)
+from scanpy.plotting import _utils
+
+
+def mixscape_barplot(
+    adata: AnnData,
+    control: str = "NT",
+    mixscape_class_global="mixscape_class_global",
+    axis_text_x_size: int = 8,
+    axis_text_y_size: int = 6,
+    axis_title_size: int = 8,
+    strip_text_size: int = 6,
+    panel_spacing_x: float = 0.3,
+    panel_spacing_y: float = 0.3,
+    legend_title_size: int = 8,
+    legend_text_size: int = 8,
+    show: bool | None = None,
+    save: bool | str | None = None,
+):
+    """Barplot to visualize perturbation scores calculated from RunMixscape function.
+
+    Args:
+        adata: The annotated data object.
+        control: Control category from the `pert_key` column. Default is 'NT'.
+        mixscape_class_global: The column of `.obs` with mixscape global classification result (perturbed, NP or NT).        show: Show the plot, do not return axis.
+        save: If True or a str, save the figure. A string is appended to the default filename. Infer the filetype if ending on {'.pdf', '.png', '.svg'}.
+
+    Returns:
+        If show is False, return ggplot object used for drawn.
+    """
+
+    count = pd.crosstab(index=adata.obs[mixscape_class_global], columns=adata.obs[control])
+    df = count / count.sum()
+    df2 = pd.melt(df, ignore_index=False).reset_index()
+    test = df2[df2[mixscape_class_global] == "KO"]
+    test = test.sort_values("value", ascending=False)
+
+    new_levels = test[control]
+    df2[control] = pd.Categorical(df2[control], categories=new_levels, ordered=False)
```

Comment:
It's not perfectly clear here what "df", "df2" and "df3" represent. Could you please name them better?